File: java-code/src/main/java/refactoring/refactoring_api/preserve_whole_object/before/HeatingPlan.java
package refactoring.refactoring_api.preserve_whole_object.before;

import java.util.ArrayList;
import java.util.List;
public class HeatingPlan {
    public static void main(String[] args) {
      Room aRoom = new Room(new TemperatureRange(10, 25));
      Plan aPlan = new Plan(); // Instantiate a Plan

      int low = aRoom.getDaysTempRange().getLow();
      int high = aRoom.getDaysTempRange().getHigh();

      if (!aPlan.withinRange(low, high)) {
        List<String> alerts = new ArrayList<>();
        alerts.add("room temperature went outside range");
      }
    }
}

File: go-code/refactoring_api/preserve_whole_object/before/heating_plan.go
package before

import "log"

type TempRange struct {
	Low, High int
}

type Room struct {
	DaysTempRange TempRange
}

type Plan struct {
	TemperatureRange TempRange
}

func (plan Plan) withinRange(low, high int) bool {
	return (low >= plan.TemperatureRange.Low) && (high <= plan.TemperatureRange.High)
}

func main() {
	aRoom := Room{DaysTempRange: TempRange{Low: 10, High: 25}}
	aPlan := Plan{} // Instantiate a Plan

	low := aRoom.DaysTempRange.Low
	high := aRoom.DaysTempRange.High

	if !aPlan.withinRange(low, high) {
		log.Fatalln("room temperature went outside range")
	}
}
